0012-BIRS-distributed-scheduler.md

```diff
@@ -0,0 +1,463 @@
+    - Things beyond basic ListJobs. This might entail things like the history of prior triggered jobs or future jobs.
+
+## Current Shortfalls
+
+The **Workflows** building block is built on top of Actor Reminders, which have scale limitation issues today. The goal is to improve the performance and scale of Actor Reminder by using the distributed scheduler.
+
+Currently, Dapr users are able to use the **Publish and Subscribe** building block, but are unable to have delayed PubSub scheduling. This scheduler  service enables users to publish a message in a future specific time , for example a week from today or a specific UTC date/time.
+
+For **Service Invocation**, this building block could also benefit from a scheduler in that it would enable the scheduling of method calls between applications.
+
+As of now, Dapr does have an **input cron binding** component, which can allow users to schedule tasks. This requires the component yaml file, where users can listen on an endpoint that is scheduled. This is limited to being an input binding only. The Scheduler Service will enable the scheduling of jobs to scale across multiple replicas, while guaranteeing at most once execution of each trigger.
+
+*Note:* Performance is the primary focus while implementing this feature given the current shortfalls.
+
+## Big Picture Idea
+
+![Big Picture](./resources/0004-BIRS-distributed-scheduler/bigPicture.png)
+
+If a user would like to store their user associated data in a specific state store of their choosing, then they can provision a state store using the Dapr State Management Building Block and set `jobStateStore ` as `true` in the state store component’s metadata section. Having the `jobStateStore` set to `true` means that their user associate data will be stored in the state store of their choosing, but their job details will still be stored in the embedded etcd. If the `jobStateStore` is not configured, then the embedded etcd will be used to store both the job details and the user associated data.
+
+*Note:* The Scheduler functionality is usable by both Standalone (Self-Hosted) and Kubernetes modes.
+
+## Implementation Details
+
+### Building Block
+
+#### Scenarios
+
+##### Example Usage
+
+Users will have a job they would like scheduled. For example, an application performs a daily backup of their database. This backup task should run every day at a specific time to ensure data integrity. The user calls to schedule their job
+using the new Dapr Scheduler Building Block.
+
+Example JSON (shown below) that you can use to schedule a job by making a request to `http://localhost:<daprPort>/v1.0/job/schedule/prd-db-backup`. This request schedules a job named `prd-db-backup` to run daily for the purpose of performing a database backup. The `@daily` schedule specification indicates that the job will run once a day, specifically at midnight (for more details, refer to the Schedule table below).
+
+```Json
+{
+  "name": "prd-db-backup",
```

Comment:
We have the name in both the path as well as the object here. I think we should only include the name in the path to be more consistent with other APIs.